EnabledProtocols is nil when using an cilium-agent which is not aware of the field definition

The commit [14ced84](https://github.com/cilium/cilium/commit/14ced84f7ef847350de8606b9e3159151a365531) have introduced two new flags `enable-ipv6-masquerade` and `enable-ipv4-masquerade` with default values (true). However, if someone makes a mistake mixing the versions of cilium-agent and cilium (cli), e.g. by applying  

```kubectl create -f https://raw.githubusercontent.com/cilium/cilium/v1.9/install/kubernetes/quick-install.yaml``` 

right after build the current master - v1.10 (from git), the running cilium-agent (v1.9) will not be aware of Masquerading.EnabledProtocols. So this field will have the default value for the type (nil).

https://github.com/cilium/cilium/blob/c603c5e378b2ecdfbc0367d85f18fe7bf1ff4fe2/api/v1/models/masquerading.go#L31

As a consequence, we can have an invalid memory access on client.go.

https://github.com/cilium/cilium/blob/c603c5e378b2ecdfbc0367d85f18fe7bf1ff4fe2/pkg/client/client.go#L408

```
cilium status
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xaebe23]
goroutine 1 [running]:
github.com/cilium/cilium/pkg/client.FormatStatusResponse(0x25db460, 0xc00057d600, 0xc0000dd130, 0x0)
	/home/vagrant/go/src/github.com/cilium/cilium/pkg/client/client.go:408 +0x7a3
github.com/cilium/cilium/cilium/cmd.statusDaemon()
	/home/vagrant/go/src/github.com/cilium/cilium/cilium/cmd/status.go:102 +0x33e
github.com/cilium/cilium/cilium/cmd.glob..func73(0x36033a0, 0x3b8c978, 0x0, 0x0)
	/home/vagrant/go/src/github.com/cilium/cilium/cilium/cmd/status.go:37 +0x25
github.com/spf13/cobra.(*Command).execute(0x36033a0, 0x3b8c978, 0x0, 0x0, 0x36033a0, 0x3b8c978)
	/home/vagrant/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x3601ea0, 0x3633800, 0x0, 0xc00006e778)
	/home/vagrant/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/vagrant/go/src/github.com/cilium/cilium/vendor/github.com/spf13/cobra/command.go:895
github.com/cilium/cilium/cilium/cmd.Execute()
	/home/vagrant/go/src/github.com/cilium/cilium/cilium/cmd/root.go:47 +0x31
main.main()
	/home/vagrant/go/src/github.com/cilium/cilium/cilium/main.go:22 +0x25
```

The fix seems to be as simple as checking the pointer before using, or just ensure it is initialized after some previous initialization.

--- a/pkg/client/client.go
+++ b/pkg/client/client.go
@@ -405,7 +405,9 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 			return "Disabled"
 		}
 
-		if !sr.Masquerading.EnabledProtocols.IPV4 && !sr.Masquerading.EnabledProtocols.IPV6 {
+		if sr.Masquerading.EnabledProtocols == nil {
+			status = enabled(sr.Masquerading.Enabled)
+		} else if !sr.Masquerading.EnabledProtocols.IPV4 && !sr.Masquerading.EnabledProtocols.IPV6 {
 			status = enabled(false)
 		} else {
 			if sr.Masquerading.Mode == models.MasqueradingModeBPF {

